store/tikv/client_batch.go

```diff
@@ -248,16 +261,25 @@ func (c *batchCommandsClient) recv() (*tikvpb.BatchCommandsResponse, error) {
 }
 
 // `failPendingRequests` must be called in locked contexts in order to avoid double closing channels.
-func (c *batchCommandsClient) failPendingRequests(err error) {
+func (c *batchCommandsClient) failPendingRequests(err error) bool {
 	failpoint.Inject("panicInFailPendingRequests", nil)
+	var (
+		heartbeatFail bool
+		i             int
+	)
 	c.batched.Range(func(key, value interface{}) bool {
 		id, _ := key.(uint64)
 		entry, _ := value.(*batchCommandsEntry)
 		entry.err = err
 		c.batched.Delete(id)
 		close(entry.res)
+		if i == 0 && entry.heartbeat {
```

Comment:
Will the `heartBeatEntry` always be the first element?
I see there is a `fetchMorePendingRequests` function.